sections/GPU.tex

```diff
@@ -1,130 +1,19 @@
-\myref{image:SM} shows an example of this, note that the ALUs and MAD corresponds to the SPs.
-The SM schedules and executes threads in groups called warps.
-A warp waits until all its threads have finished executing, before the next instruction is fetched. 
-
-
-%http://www.cs.virginia.edu/~skadron/Papers/thesis_prateeksha.pdf
-\subsection{Conclusion} %This needs serious rework - but not now!
-
-The architectural difference of the GPU from the CPU, are intended to maximise the throughput for problems that can handle a higher latency.
-The lack of control and cache memory for the processing units used in the GPU, also greatly reduces its capabilities of handling complex issues that requires a concurrent approach.
-Since the GPU does not have interupts an instruction in a given thread will run until it is complete and no thread will start without all the available resources.
-Thus some problems will be ill suited for the GPU but any problems that can be divided into smaller blocks of data, and designated to run on different threads, will benefit substantially from the increased computing power the GPU provides.
-The scheduling of said blocks are controlled by the hardware and no explicit management is required.
-In conclusion due to the magnitude of threads available in a GPU, its memory hierarchy for said threads and the ability of the hardwares to manage those threads, any problem with data parallel workloads will be executed significantly faster than on any equivalent CPU.
 
-%First Draft
-%The result of this architecture is increased parallel computing power through the sacrifice of sequential computing power.
-%The GPU sacrifices the cache memory of the CPU, in exchange for more computing power.
-%This results in a very high throughput, work done in an amount of time, given that there is enough work.
-%It also results in higher latency, the time from which an instruction is initiated untill the effects have taken place.
-%Is is the direct opposite of a CPU.
-%In a CPU low latency is preferred, the latency is hidden through its ability to do work concurrently the low latency is achieved by the caches, something which the GPU sacrificed.
-%Although with low latency, also follows low throughput, which is why the GPU is more optimized for heavy computational work that can be done in parallel.
-%The structure of SMs, and the work distribution of the GPU, is what enables it to use the SIMD model.
-%The SM follows only one set of instructions but on multiple threads of data, the threads are managed and scheduled by the hardware itself.
-%In conlusion, the GPU architecture allows us to efficiently compute any workload which can be done using parallel computing, and can tolerate the higher latency.
+A basic layout comparison of a Central Processing Unit (``CPU'') and a GPU is shown in \myref{image:GPUCPUimage}.
```

Comment:
Strictly not correct at is not a picture of an actual CPU and GPU, but an image representing their transistor allocation.  "A representational comparison of the CPU and GPU transistor usage [...]"